h5netcdf/core.py

```diff
@@ -234,6 +243,29 @@ def __init__(self, parent, name):
                     if k.startswith('_nc4_non_coord_'):
                         var_name = k[len('_nc4_non_coord_'):]
                     self._variables.add(var_name)
+
+        # One last pass to get the current sizes.
+        #
+        # I'm not entirely sure how the netcdf C api determines the current
+        # size - should probably be looked into.
```

Comment:
I'm actually not sure we need or want section. As long as resizing is done consistently across all variables (necessary to ensure a valid netCDF file), just looking at the dimension scale should be enough.